File: MyApp.API/Models/Asset.cs
using System;
using System.Collections.Generic;

namespace MyApp.API.Models
{
    public class Asset
    {
        public int Id { get; set; }
        public int ClientId { get; set; }
        public virtual Client Client { get; set; }
        public int TenantId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Location { get; set; }
        public int Quantity { get; set; }
        public DateTime ManufacturedDate { get; set; }
        public DateTime? InstalledDate { get; set; }
        public int ProductId { get; set; }
        public virtual Product Product { get; set; }

        public virtual ICollection<MaintScheduleAsset> MaintScheduleAssets { get; set; }

            // Audit Fields
        public bool? IsActive { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public DateTime? DeletedOn { get; set; }
        public int CreatedBy { get; set; }
        public int UpdatedBy { get; set; }
        public int DeletedBy { get; set; }

    }
}

File: MyApp-SPA/src/app/_models/asset.ts

export interface Asset{
  id: number;
  tenantId: number;
  location: string;
  quantity: number;
  manufacturedDate: string;
  product: ProductInAsset;
  client: ClientInAsset;
  inspections: Inspection[];
  isActive: boolean;
  isDeleted: boolean;
  createdOn: string;
  updatedOn: string;
  deletedOn?: any;
  createdBy: number;
  updatedBy: number;
  deletedBy: number;
}

interface Inspection {
  assetId: number;
  maintScheduleId: number;
  productCategoryId: number;
  nextInspectionDate: string;
  name: string;
  description: string;
  monthsInterval: number;
  lastInspectionDate: string;
  calculatedNextInspectionDate: string;
  lastResult: string;
  isActive: boolean;
}

interface ClientInAsset {
  id: number;
  tenantId: number;
  company: string;
  nif: string;
  jobTitle: string;
  adress: string;
  city: string;
  state_Province: string;
  ziP_PostalCode: string;
  country: string;
  email: string;
  phone: string;
  observations: string;
}

interface ProductInAsset {
  id: number;
  name: string;
  description: string;
  productCategoryId: number;
  productCategoryName: string;
}
